### ISSUE TITLE
`toArray` in REPL does not show ellipses for structural records

### ISSUE THREAD

**Issue by boogerlad** [lamdu/lamdu#133]
<img width="639" alt="Screen Shot 2021-11-29 at 1 43 02 AM" src="https://user-images.githubusercontent.com/1903750/143820558-086680db-c425-44a6-9776-61c994249712.png">

For plain values, ellipses are shown, as expected
<img width="906" alt="Screen Shot 2021-11-29 at 1 43 26 AM" src="https://user-images.githubusercontent.com/1903750/143820606-6fd1fcb9-92e3-49b0-8eb3-9851dd58638c.png">

On a side note, how can I see all the values? (up to 70, in this case)

**Comment by Peaker** [comment]
The ellipsis here are due to the laziness of the list's continuation not being computed by default (hidden behind a lambda) rather than the record itself.

Once converted to an array, there's no ellipsis anymore, but the array is truncated by the REPL view with no indication of truncation and no ellipsis, which should be improved.

Showing all the elements can be a bit huge, perhaps we need to allow internal scroll inside such table views?

**Comment by boogerlad** [comment]
> Showing all the elements can be a bit huge, perhaps we need to allow internal scroll inside such table views?
Yes. I think this is a fantastic idea. The biggest annoyance that I have with debug prints is when there's too much data, it makes the other debug prints less visible.

> the array is truncated by the REPL view with no indication of truncation and no ellipsis
in the first screenshot, the overall output shows
```
dos | uno
__________
0   | 1
1   | 1
1   | 2
2   | 3
3   | 5
```
with NO ellipses.

However, the second screenshot shows `[1,1,2,3,5,8,13,24,34,55,...]` which does have ellipses, so are you sure this bug has nothing to do with the record?

**Comment by Peaker** [comment]
> > Showing all the elements can be a bit huge, perhaps we need to allow internal scroll inside such table views?
> 
> Yes. I think this is a fantastic idea. The biggest annoyance that I have with debug prints is when there's too much data, it makes the other debug prints less visible.
> 
> > the array is truncated by the REPL view with no indication of truncation and no ellipsis
> 
> in the first screenshot, the overall output shows
> 
> ```
> dos | uno
> __________
> 0   | 1
> 1   | 1
> 1   | 2
> 2   | 3
> 3   | 5
> ```
> 
> with NO ellipses.
> 
> However, the second screenshot shows `[1,1,2,3,5,8,13,24,34,55,...]` which does have ellipses, so are you sure this bug has nothing to do with the record?

It does have something to do with records - namely the table-sugar used for "array of records" does truncation with no ellipsis, whereas "array of numbers" does truncation with ellipsis.

We need to fix "array of records" table view to truncate gracefully, or support scrolling

**Comment by yairchu** [comment]
(and to support scrolling we would need to transition to a graphics library that has clipping, which is already in our plans)
